packages/common/src/search/wellKnownCatalog.ts

```diff
@@ -0,0 +1,268 @@
+import { IUser } from "@esri/arcgis-rest-types";
+import { getFamilyTypes } from "../content";
+import { EntityType, IHubCatalog, IHubCollection } from "./types";
+
+export type WellKnownCatalog =
+  | "myContent"
+  | "myGroup"
+  | "favorites"
+  | "organization"
+  | "world";
+
+export type WellKnownCollection =
+  | "appAndMap"
+  | "content"
+  | "dataset"
+  | "document"
+  | "event"
+  | "feedback"
+  | "initiative"
+  | "project"
+  | "site"
+  | "solution"
+  | "template";
+
+/** This function returns a single IHubCatalog definition JSON object based on the
+ * name and entity type requested
+ * @param i18nScope Translation scope to be interpolated into the catalog
+ * @param name Name of the catalog requested
+ * @param entityType
+ * @param user Owner of the entity, optional but required for certain catalogs
+ * @param collectionNames A list of collection names, optional, if passed in,
+ * only those collections in the catalog will be returned
+ */
+export function getWellKnownCatalog(
+  i18nScope: string,
+  name: WellKnownCatalog,
+  entityType: EntityType,
+  user?: IUser,
+  collectionNames?: WellKnownCollection[]
+): IHubCatalog {
+  switch (entityType) {
+    case "item":
+      return getWellknownItemCatalog(i18nScope, name, user, collectionNames);
+    /* Add other entity handlers here, e.g. getWellknownEventCatalog */
+    default:
+      throw new Error(`Wellknown catalog not implemented for "${entityType}"`);
+  }
+}
+
+/**
+ * @param i18nScope Translation scope to be interpolated into the catalog
+ * @param name Name of the catalog requested
+ * @param user Owner of the entity, optional but required for certain catalogs
+ * @param collectionNames A list of collection names, optional, if passed in,
+ * only those collections in the catalog will be returned
+ * @returns An Item IHubCatalog definition JSON object
+ */
+function getWellknownItemCatalog(
+  i18nScope: string,
+  name: WellKnownCatalog,
+  user?: IUser,
+  collectionNames?: WellKnownCollection[]
+): IHubCatalog {
+  // Check if i18nScope is defined, if so add a `.` on the end
+  if (i18nScope !== "") {
+    i18nScope = `${i18nScope}.`;
+  }
+  let catalog;
+  switch (name) {
+    case "myContent":
+      if (user) {
+        catalog = {
+          schemaVersion: 1,
+          title: `{{${i18nScope}catalog.myContent:translate}}`,
```

Comment:
My only problem with this is that it is specific to ember interpolation. Is there any reason that we can't just have a function in ember for obtaining the translated name of a given catalog?

Reply:
We can, but that adds more complexity IMO. With the strings wrapped in the double curlies and `:translate` at the end, it automatically gets picked up by `interpolate` and we can use this `interpolateTranslations` function [interpolateTranslations](https://github.com/Esri/hub-components/pull/974) in both hub-components and opendata.